sample/TKAccessSamples.m

```diff
@@ -210,13 +210,33 @@ -(Money*)carefullyUse:(TKMember*)grantee
             [accountIds addObject:accounts[i].id];
         }
     }
+
     if (accountIds.count < 1) {
+        // We have access to no accounts; return zero balance
         return [Money new];
     }
-    Money *balance0 = [granteeSync getBalance:accountIds[0]
-                                      withKey:Key_Level_Standard].available;
-    [granteeSync clearAccessToken];
-    return balance0;
+    
+    for (int i = 0; i < accountIds.count; i++) {
+        @try {
+            Money *balance = [granteeSync getBalance:accountIds[i]
+                                             withKey:Key_Level_Standard].available;
+            [granteeSync clearAccessToken];
+            return balance;
+        }
+        @catch (NSError *error) {
```

Comment:
I looked for examples of the idiomatic way to catch-and-maybe-rethrow in Objective-C, but didn't find any. Be skeptical, brave reviewers.